File: server/src/main/java/com/krillsson/sysapi/config/CacheConfiguration.java
package com.krillsson.sysapi.config;

import java.util.concurrent.TimeUnit;

public class CacheConfiguration {
    private long duration;
    private TimeUnit unit;

    public CacheConfiguration(long duration, TimeUnit unit) {
        this.duration = duration;
        this.unit = unit;
    }

    public CacheConfiguration() {
    }

    public long getDuration() {
        return duration;
    }

    public TimeUnit getUnit() {
        return unit;
    }
}

File: server/src/main/java/com/krillsson/sysapi/config/CacheConfiguration.kt
package com.krillsson.sysapi.config

import com.fasterxml.jackson.annotation.JsonProperty
import java.util.concurrent.TimeUnit

data class CacheConfiguration(
    @JsonProperty val enabled: Boolean = true,
    @JsonProperty val duration: Long = 5,
    @JsonProperty val unit: TimeUnit = TimeUnit.SECONDS
)